src/sbt-test/sbt-buildinfo/task/build.sbt

```diff
@@ -0,0 +1,33 @@
+import StableState.{ counterOutOfTaskGraph, counterInTaskGraph }
+
+scalaVersion in ThisBuild := "2.12.4"
+
+val projOutOfTaskGraph1 = project settings (
+  sourceGenerators in Compile += Def.task { counterOutOfTaskGraph.incrementAndGet(); Nil }.taskValue
+)
+
+val projOutOfTaskGraph2 = project dependsOn projOutOfTaskGraph1 settings (
+  BuildInfoPlugin.buildInfoDefaultSettings,
+  addBuildInfoToConfig(Test),
+  buildInfoKeys in Test += fullClasspath in Compile
+)
+
+val projInTaskGraph1 = project settings (
+  sourceGenerators in Compile += Def.task { counterInTaskGraph.incrementAndGet(); Nil }.taskValue
+)
+
+val projInTaskGraph2 = project dependsOn projInTaskGraph1 settings (
+  BuildInfoPlugin.buildInfoDefaultSettings,
+  addBuildInfoToConfig(Test),
+  buildInfoKeys in Test += (fullClasspath in Compile).taskValue
```

Comment:
In https://github.com/sbt/sbt/pull/2943 I'm injecting `.taskValue` as macro when the LHS is `Initialize[Seq[Task[...]]]`.
Would you be able to take advantage of that if `buildInfoKeys` were `SettingKey[Seq[Task[String]]`, and you unify other miscellaneous build info entries to a `Task[...]` instance?

Reply:
I'm not sure - maybe. But how are you going to change the type of `buildInfoKeys`?

Reply:
We can just change the type and bump up minor version.

Reply:
What about any plugin that depends on sbt-buildinfo?

Reply:
I am personally unaware of any such plugin, and I don't feel obligation to keep bincompat as long as we bump to 0.8.0.

Reply:
I don't think this is going to work because then this no longer works:

Reply:
or (and this was the same argument as sbt/sbt#2943) own the fact that `TaskKey`s and `Task` values are different types in sbt, and there are different situations for using either.

Reply:
Perhaps:

Reply:
Perhaps: